src/mpags-cipher.cpp

```diff
@@ -92,19 +95,9 @@ int main(int argc, char* argv[])
 
     std::string outputText;
 
-    switch (settings.cipherType) {
-        case CipherType::Caesar: {
-            // Run the Caesar cipher (using the specified key and encrypt/decrypt flag) on the input text
-            CaesarCipher cipher{settings.cipherKey};
-            outputText = cipher.applyCipher(inputText, settings.cipherMode);
-            break;
-        }
-        case CipherType::Playfair: {
-            PlayfairCipher cipher{settings.cipherKey};
-            outputText = cipher.applyCipher(inputText, settings.cipherMode);
-            break;
-        }
-    }
+    // create Cipher object and apply the cipher to get the de/encoded text
+    auto cipher{cipherFactory(settings.cipherType, settings.cipherKey)};
+    outputText = cipher->applyCipher(inputText, settings.cipherMode);
```

Comment:
You need to check that the pointer is valid before using it.
```suggestion
    auto cipher{cipherFactory(settings.cipherType, settings.cipherKey)};
    if ( ! cipher ) {
      std::cerr << "[error] invalid pointer returned from cipher factory" << std::endl;
      return 1;
    }
    outputText = cipher->applyCipher(inputText, settings.cipherMode);
```